src/bin/wasm-tools/objdump.rs

```diff
@@ -70,40 +84,91 @@ impl Opts {
     }
 }
 
+#[derive(Default, Debug)]
+struct IndexSpace {
+    modules: Option<u32>,
```

Comment:
Do these need to be `Option`? What if we didn't increment these values upon encountering a module or component section, but instead in `end` (i.e. we've just finished a nested module/component, increment the parent index space's counters)? That would also simplify the increment logic.

Reply:
Yeah, agreed that this is better, I also defer calling `printer.start` when encountering a `Component` or `Module` rather than when encountering a `Version`, that makes it easier to handle the "top-level" case.